pkg/indexer/localcredentials.go

```diff
@@ -0,0 +1,103 @@
+package indexer
+
+import (
+	"context"
+	"reflect"
+
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api"
+)
+
+type LocalCredentialIndexer struct {
+	obj    client.Object
+	name   string
+	logger *zap.SugaredLogger
+}
+
+// Reconciliable is implemented by CRD objects used by indexes to trigger reconciliations
+type Reconciliable interface {
+	ReconciliableRequests() []reconcile.Request
+}
+
+// ReconciliableList is a Reconciliable that is also a CRD list
+type ReconciliableList interface {
+	client.ObjectList
+	Reconciliable
+}
```

Comment:
Nobody is implementing or relying on them, why do we need them?

Reply:
Checking, I might have overlook to cleanup. Thanks for the heads up!